BlazorForms/Final/BlazorForms/Data/Contact.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.Diagnostics.CodeAnalysis;
+
+namespace BlazorForms.Data
+{
+    public class Contact
+    {
+        [Required(ErrorMessage = "Please provide a first name.")]
+        public string FirstName { get; set; } = string.Empty;
+
+        [Required]
+        public string LastName { get; set; } = string.Empty;
+        public string? PhoneNumber { get; set; }
+
+        private string? _Email { get; set; }
```

Comment:
Typically, we only use the underscore prefix with fields. This looks like it is expected to be the backing field for the `Email` property so I would just make it a field rather than a property.
```suggestion
        private string? _Email;
```

You will also want to make sure the casing between other fields is the same. For example, `ErrorModel._logger` is using camel case vs this one using pascal case.